send-transaction-service/src/send_transaction_service.rs

```diff
@@ -298,9 +308,17 @@ impl SendTransactionService {
                     {
                         // take a lock of retry_transactions and move the batch to the retry set.
                         let mut retry_transactions = retry_transactions.lock().unwrap();
-                        let transactions_to_retry = transactions.len();
+                        let mut transactions_to_retry: usize = 0;
                         let mut transactions_added_to_retry: usize = 0;
                         for (signature, mut transaction_info) in transactions.drain() {
+                            // drop transactions with 0 max retries
+                            let max_retries = transaction_info
+                                .get_max_retries(default_max_retries, service_max_retries);
+                            if max_retries == Some(0) {
```

Comment:
we should probably replace the `Option` here with a custom enum with  variant names that self-document. we want something like
```rust
struct MaxRetries {
    Infinite,
    NumRetries(usize),
}
```